pageserver/pagebench/src/cmd/getpage_latest_lsn.rs

```diff
@@ -116,31 +117,26 @@ async fn main_impl(
                     .keyspace(timeline.tenant_id, timeline.timeline_id)
                     .await?;
                 let lsn = partitioning.at_lsn;
-
-                let ranges = partitioning
-                    .keys
-                    .ranges
-                    .iter()
-                    .filter_map(|r| {
-                        let start = r.start;
-                        let end = r.end;
-                        // filter out non-relblock keys
-                        match (is_rel_block_key(&start), is_rel_block_key(&end)) {
-                            (true, true) => Some(KeyRange {
-                                timeline,
-                                timeline_lsn: lsn,
-                                start: start.to_i128(),
-                                end: end.to_i128(),
-                            }),
-                            (true, false) | (false, true) => {
-                                unimplemented!("split up range")
-                            }
-                            (false, false) => None,
+                let mut filtered = KeySpaceAccum::new();
+                // let's hope this is inlined and vectorized...
+                // TODO: turn this loop into a is_rel_block_range() function.
+                for r in partitioning.keys.ranges.iter() {
+                    let mut i = r.start;
+                    while i != r.end {
+                        if is_rel_block_key(&i) {
+                            filtered.add_key(i);
                         }
-                    })
-                    .collect::<Vec<_>>();
-
-                anyhow::Ok(ranges)
+                        i = i.next();
+                    }
+                }
```

Comment:
Surprisingly the runtime overhead doesn't seem terrible with 200 tenants, total 4.7 GiB of layers.

Will try again with 20k tenants and several hundred GiB of layers.

Reply:
Ok, created a setup with 180k layer files (20k tenants * 9 layers). 
Total physical size is 463GiB.

For 5k tenants, the range filtering takes 0.6seconds.
That's a tiny fraction of the overall time it takes for pagebench to get ready to send requests.
So, this is good enough for now.